earn/src/data/MarginAccount.ts

```diff
@@ -111,95 +119,147 @@ export type UniswapPoolInfo = {
-      } catch (e) {
-        console.error(`borrowerLens.getAssets failed for account ${accountAddress} in pool ${uniswapPool}`, e);
-        return null;
-      }
-
-      const health = healthData[0].lt(healthData[1]) ? healthData[0] : healthData[1];
-      const assets: Assets = {
-        token0Raw: Big(assetsData.fixed0.toString())
-          .div(10 ** token0.decimals)
-          .toNumber(),
-        token1Raw: Big(assetsData.fixed1.toString())
-          .div(10 ** token1.decimals)
-          .toNumber(),
-        uni0: Big(assetsData.fluid0C.toString())
-          .div(10 ** token0.decimals)
-          .toNumber(),
-        uni1: Big(assetsData.fluid1C.toString())
-          .div(10 ** token1.decimals)
-          .toNumber(),
-      };
-      const liabilities: Liabilities = {
-        amount0: Big(liabilitiesData.amount0.toString())
-          .div(10 ** token0.decimals)
-          .toNumber(),
-        amount1: Big(liabilitiesData.amount1.toString())
-          .div(10 ** token1.decimals)
-          .toNumber(),
-      };
-      return {
-        address: accountAddress,
-        uniswapPool,
-        token0,
-        token1,
-        feeTier,
-        assets,
-        liabilities,
-        health: health.div(1e9).toNumber() / 1e9,
-      };
-    }
-  );
-  return (await Promise.all(marginAccounts)).filter((account) => account !== null) as MarginAccountPreview[];
+  let contractCallContext: ContractCallContext[] = [];
+
+  marginAccountsAddresses.forEach(({ address: accountAddress, uniswapPool }) => {
+    const uniswapPoolInfo = uniswapPoolDataMap.get(`0x${uniswapPool}`) ?? null;
+
+    if (uniswapPoolInfo === null) return;
+
+    const token0 = uniswapPoolInfo.token0;
+    const token1 = uniswapPoolInfo.token1;
+    const feeTier = NumericFeeTierToEnum(uniswapPoolInfo.fee);
+
+    if (!token0 || !token1) return;
+
+    contractCallContext.push({
+      reference: accountAddress,
+      contractAddress: ALOE_II_BORROWER_LENS_ADDRESS,
+      abi: MarginAccountLensABI,
+      calls: [
+        {
+          reference: 'getAssets',
+          methodName: 'getAssets',
+          methodParameters: [accountAddress],
+        },
+        {
+          reference: 'getLiabilities',
+          methodName: 'getLiabilities',
+          methodParameters: [accountAddress, true],
+        },
+        {
+          reference: 'getHealth',
+          methodName: 'getHealth',
+          methodParameters: [accountAddress, true],
+        },
+      ],
+      context: {
+        feeTier: feeTier,
+        token0: token0,
+        token1: token1,
+        accountAddress: accountAddress,
+        uniswapPool: uniswapPool,
+      },
+    });
+  });
+
+  const results = (await multicall.call(contractCallContext)).results;
+
+  let marginAccounts: MarginAccountPreview[] = [];
+
+  Object.values(results).forEach((value) => {
+    const contractResults = value.callsReturnContext;
+    const updatedReturnContext = convertBigNumbers(contractResults);
```

Comment:
It seems like `convertBigNumbers` should operate on `contractResults.returnValues` instead of the entire `contractResults` object.